### PR TITLE
Room members: gray out the expired users

### PR DESCRIPTION
**Pull request by yostyle**
#273

### CODE REVIEW

File: Tchap/Managers/User/UserService.swift [hunk 1]
```diff
@@ -109,21 +109,14 @@ final class UserService: NSObject, UserServiceType {
...
-    func isAccountExpired(for userId: String, completion: @escaping ((MXResponse<Bool>) -> Void)) -> MXHTTPOperation? {
-        return self.getUserInfo(for: userId) { (response) in
-            switch response {
-            case .success(let userInfo):
-                completion(.success(userInfo.expired))
+            case .success(let usersInfo):
+                if let deactivated = usersInfo[userId]?.deactivated {
+                    completion(.success(deactivated))
+                } else {
+                    completion(.success(false))
```
Comment (giomfo) [inline Tchap/Managers/User/UserService.swift:118]: I would prefer consider this case as a failure, don't you think so?
```suggestion
                    completion(.failure(UserServiceError.unknown))
```

File: Tchap/Modules/Room/Members/RoomParticipantsViewController.m [hunk 2]
```diff
@@ -902,36 +902,39 @@ - (void)checkExpiredAccounts:(void (^)(void))completion
     dispatch_group_t requestsGroup = dispatch_group_create();
```
Comment (giomfo) [inline Tchap/Modules/Room/Members/RoomParticipantsViewController.m:902]: You should remove this dispatch_group_t, it is useless now

File: Tchap/Modules/Room/Members/RoomParticipantsViewController.m [hunk 3]
```diff
@@ -902,36 +902,39 @@ - (void)checkExpiredAccounts:(void (^)(void))completion
...
     }
-    for (Contact *contact in invitedParticipants)
-    {
-        if (contact.mxMember.userId)
+    
+    dispatch_group_enter(requestsGroup);
+    op = [self.userService getUsersInfoFor:contactIds success:^(NSDictionary<NSString *,id> * _Nonnull usersInfo) {
+        for (Contact *contact in participants)
+        {
+            UserStatusInfo *userInfo = [usersInfo objectForKey:contact.mxMember.userId];
```
Comment (giomfo) [inline Tchap/Modules/Room/Members/RoomParticipantsViewController.m:920]: `contact.mxMember` is nullable, you should check before using `contact.mxMember.userId` as a key

To get this kind of participant, invite an email (for which no account exists yet) in the room. Then `contact.mxMember` is null

**Review by giomfo — COMMENTED** [review]
- Commented
- Additional comment: I would keep in UserService the swift version of `isAccountExpired` method. I agree this is not used for the moment, but this may be useful to have this shortcut later

File: Tchap/Managers/User/UserService.swift (hunk 1 above)
Comment (yostyle) [reply Tchap/Managers/User/UserService.swift]: Applied

File: Tchap/Modules/Room/Members/RoomParticipantsViewController.m (hunk 2 above)
Comment (yostyle) [reply Tchap/Modules/Room/Members/RoomParticipantsViewController.m]: removed

File: Tchap/Modules/Room/Members/RoomParticipantsViewController.m (hunk 3 above)
Comment (yostyle) [reply Tchap/Modules/Room/Members/RoomParticipantsViewController.m]: Applied

**Review by giomfo — APPROVED** [review]

**Comment by giomfo** [comment]
@yostyle please review my commits, I let you merge if you agree with them

**Comment by yostyle** [comment]
LGTM

**Pull request merged**